arch/src/main/java/knaufdan/android/arch/network/Repository.kt

```diff
@@ -0,0 +1,60 @@
+package knaufdan.android.arch.network
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import knaufdan.android.arch.base.IGenericType
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+abstract class Repository<ServiceType>(baseUrl: String) : IGenericType<ServiceType> {
```

Comment:
- Rename to BaseService 
- Pass in a `ServiceConfig` object which is easer to expand in the future and can hold specification via RetroFit interface

Reply:
done.